Fix 72

huard opened the pull request:
## Overview

This PR fixes #72 

Changes:

* Added support for passing model parameters directly, without having to specify which rv file it belongs to. 
* Added default temp directory when initializing Raven class. 
* Added docstrings

huard commented on raven/models/basic.py:
```diff
@@ -23,7 +23,7 @@
 import numpy as np
 
-raven_exec = Path(raven.__file__).parent.parent / 'bin' / 'raven'
+raven_exec = str(Path(raven.__file__).parent.parent / 'bin' / 'raven')
```
Good catch. I'm guessing it's installed by another library we're using.

huard replied:
This really got me wishing we had dropped python 2.7 support...

Zeitsperre replied:
To be fair, the changes I've added to the regionalisation branch specify that the make install use python3.6. The code is already 99% compatible with python3.

huard commented:
I'll need a review on this. 
I simplified the overwrite portion and fixed 2.7 bugs that have lingered until now (I don't understand why the test suite did not pick them up months ago.)

Zeitsperre commented on tests/common.py:
```diff
@@ -19,7 +19,7 @@
 
 TESTS_HOME = Path(__file__).parent
```
Another case of `pathlib` usage.

Zeitsperre replied:
Perhaps adding the `pathlib` library would make it so that you don't need to cast paths to strings?

huard replied:
The issue is that Python 2.7 does not `open` Path objects. To be clear, the pathlib library seems to be installed, even though it's not explicitly in the environment file, otherwise there would be an ImportError when I import pathlib. 

Since we're going to move to Py3 where pathlib is built-in, I suggest we use `str` for now and remove them when we drop Py2 support.
